android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/TranslationOperation.kt

```diff
@@ -25,6 +25,11 @@ enum class TranslationOperation {
      */
     FETCH_SUPPORTED_LANGUAGES,
 
+    /**
+     * The list of downloaded languages that the translation engine should fetch.
+     */
+    FETCH_DOWNLOADED_LANGUAGES,
```

Comment:
I thought this fetches the already `downloaded` languages, what did you mean by `downloadable`?

Besides that, it sounds like we want to distinguish between `Language` versus the state that would eventually be returned with the update/delete changes, like `FETCH_DOWNLOADED_LANGUAGE_MODEL_STATES` and `requestDownloadedLanguageModelStates`?

Reply:
>I thought this fetches the already downloaded languages, what did you mean by downloadable?

This is more the state of the download condition of the models, rather than a list of the already downloaded models. It is the data for this [Figma](https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7202%3A115491&mode=dev). Or in other words, the download state can be [true or false](https://searchfox.org/mozilla-mobile/rev/1dc5f1289afb580e32b66303a78ab5d92ce30ca5/firefox-android/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/translate/LanguageModel.kt#20). 100% splitting hairs here and open to any name!

>Besides that, it sounds like we want to distinguish between Language versus the state that would eventually be returned with the update/delete changes, like FETCH_DOWNLOADED_LANGUAGE_MODEL_STATES and requestDownloadedLanguageModelStates?

Yeah, preventing naming collisions is tricky. 😵‍💫 I feel like I've been having more trouble than usual with naming for translations. Maybe the update condition is `UPDATE_LANGUAGE_MODEL` or `MANAGE_LANGUAGE_MODEL` (or something?), I think the way the [install/delete API works is that it can be both](https://searchfox.org/mozilla-central/rev/9cd4ea81e27db6b767f1d9bbbcf47da238dd64fa/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/TranslationsController.java#135).

Reply:
Ah I see, thanks for explaining that! I updated to FETCH_LANGUAGE_MODELS.